### ISSUE TITLE
Restore "Connect with"

### ISSUE THREAD

**Issue by lmoureaux** [longturn/freeciv21#889]
**Describe the bug**

The comparatively minor "connect with" features were dropped in #824 and do nothing. Restore them.

**Comment by lmoureaux** [comment]
Summary of a discussion today on Discord:

* We'd be fine with building the fastest built road and not the shortest as was previously done
* No change to the cost function required, everything we need is already there

Still, we need two things before we start thinking about implementing this:

1. A callback during TC simulation that can insert orders (how to build road then rails on top?). We already have one to reject vertices (e.g. no paradropping when building roads).
2. A callback that can modify the current tile when making a probe (to add the roads just built by our worker).

**Comment by 3dsf** [comment]
Am I understanding correctly that the connect to function with roads/irrigation is not available currently?

**Comment by jwrober** [comment]
Yes. We did a big refactor of the go-to code and this is left to be fixed.

**Comment by 3dsf** [comment]
@jwrober ,
thanks, I was just unsure if was  an issue with my client

**Comment by lmoureaux** [comment]
I think #2499 is the way to go and it will be easier to implement after revising map interactions for https://github.com/longturn/freeciv21/milestone/7
